packages/react-components/src/components/FeedbackDialog/index.tsx

```diff
@@ -155,15 +183,17 @@ export const FeedbackDialog = ({
             name="feedback-dialog"
             onSubmit={(e: React.BaseSyntheticEvent) => {
               e.preventDefault();
-              const errors = validateField(
+              const errors = validateFields(
                 schema,
                 { message: e.target.message.value },
-                "message"
+                ["email", "message"]
               );
               if (Object.keys(errors).length === 0) {
                 onSubmit({
+                  email: withEmailInput ? e.target.email.value : undefined,
                   message: e.target.message.value,
                 });
+                setOpen(false);
```

Comment:
i think `Props['onSubmit']` should require a Promise. This way `setOpen(false)` could be closed only when `onSubmit` resolves successfully.

In case of error, the dialog would still be open and user could see error message